core/modules/node/node.theme.inc

```diff
@@ -161,6 +161,32 @@ function theme_node_admin_overview($variables) {
  * @see node.tpl.php
  */
 function template_preprocess_node(&$variables) {
+  if ($variables['elements']['#node']->status == NODE_NOT_PUBLISHED) {
```

Comment:
I feel like this should continue to be in basis_preprocess_node(), as it contains specific markup and themes need to style it separately.

Reply:
I feel like the functionality should be available to you regardless of the theme you use; thus why I moved it, but I'm not passionate about it.